src/Modules/OrchardCore.Commerce/Endpoints/Services/ShoppingCartService.cs

```diff
@@ -77,11 +76,11 @@ public async Task<string> AddItemAsync(ShoppingCartLineUpdateModel line, string
         }
         catch (FrontendException ex)
         {
-            var errors = ex.HtmlMessages;
-            errored = errors.ConvertLocalizedHtmlStringList();
+            var errors = ex.HtmlMessages.Select(error => error.Html());
+            strError = string.Join(System.Environment.NewLine, errors);
```

Comment:
Why do you need `Environment.NewLine` here? This is a web API, so the server's newline style shouldn't affect the response. The endpoint should always return `\n` to the client, regardless whether the server is on Windows Server or Linux.

```suggestion
            return string.Join('\n', errors);
```